src/emulator/module/include/module/lay_out_args.h

```diff
@@ -47,23 +48,39 @@ constexpr std::tuple<ArgLayout, LayoutArgsState> add_to_gpr_or_stack(const Layou
     const size_t gpr_alignment = gpr_required;
     const size_t gpr_index = align(state.gpr_used, gpr_alignment);
     const size_t next_gpr_used = gpr_index + gpr_required;
-    
+
     // Does variable not fit in register file?
     if (next_gpr_used > 4) {
         return add_to_stack<Arg>(state);
     }
-    
+
     // Lay out in registers.
     const ArgLayout layout = { ArgLocation::gpr, gpr_index };
-    const LayoutArgsState next_state = { next_gpr_used, state.stack_used };
-    
+    const LayoutArgsState next_state = { next_gpr_used, state.stack_used, state.float_used };
+
+    return { layout, next_state };
+}
+
+template <typename Arg>
+constexpr std::tuple<ArgLayout, LayoutArgsState> add_to_float(const LayoutArgsState &state) {
+    const size_t float_index = state.float_used;
+    const size_t next_float_used = state.float_used + 1;
```

Comment:
No time to properly review but floats probably run out of registers and need to spill onto stack like integers do.

Reply:
Doc says that they will use fp registers and I see a lot of them. S0-S31

Reply:
I've also checked that on assembly code for sceGxmSetViewport and it is all registers